services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -0,0 +1,219 @@
+            northboundService.getAllMeters(it.dpId)
+        }
+
+        then: "Only the default meters should be present on each switch"
+        meters.each {
+            assert it.meterEntries.size() == 2
+            assert it.meterEntries.every { Math.abs(it.rate - (DISCO_PKT_RATE * DISCO_PKT_SIZE) / 1024L) <= 1 }
+            assert it.meterEntries.every { it.meterId <= Constants.MAX_SERVICE_METER_ID }
+            assert it.meterEntries.every { ["KBPS", "BURST", "STATS"].containsAll(it.flags) }
+        }
+    }
+
+    def "Default meters should be present on all switches"() {
+        given: "All Openflow 1.3 compatible switches"
+        def switches = getNonCentecSwitches()
+
+        when: "Try to get meters from all the switches"
+        List<SwitchMeterEntries> meters = switches.collect {
+            northboundService.getAllMeters(it.dpId)
+        }
+
+        then: "Only the default meters should be present on each switch"
+        meters.each {
+            assert it.meterEntries.size() == 2
+            assert it.meterEntries.every { it.rate == DISCO_PKT_RATE }
+            assert it.meterEntries.every { it.burstSize == 0 }
+            assert it.meterEntries.every { it.meterId <= Constants.MAX_SERVICE_METER_ID }
+            assert it.meterEntries.every { ["PKTPS", "BURST", "STATS"].containsAll(it.flags) }
+        }
+    }
+
+    def "Creating a flow should create new meters on a switch"() {
+        given: "A switch with OpenFlow 1.3 support"
+        def sw = topology.getActiveSwitches().find { it.ofVersion == "OF_13" }
+
+        when: "Get default meters from the switch"
+        def defaultMeters = northboundService.getAllMeters(sw.dpId)
+        assert defaultMeters
+
+        and: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+
+        then: "New meters should appear after flow setup"
+        def newMeters = northboundService.getAllMeters(sw.dpId)
+        def newMeterEntries = newMeters.meterEntries.findAll { !defaultMeters.meterEntries.contains(it) }
+        newMeterEntries.size() == 2
+
+        and: "All new meters should have KBPS, BURST and STATS flags installed"
+        newMeterEntries.every { it.flags.sort().equals(["KBPS", "BURST", "STATS"].sort()) }
+
+        and: "All new meters rate should be equal to flow's rate"
+        newMeterEntries*.rate.every { it == flow.maximumBandwidth }
+
+        when: "Delete a flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "New meters should disappear from the switch"
+        def newestMeters = northboundService.getAllMeters(sw.dpId)
+        newestMeters.meterEntries.containsAll(defaultMeters.meterEntries)
+        newestMeters.meterEntries.size() == defaultMeters.meterEntries.size()
+    }
+
+    @Unroll
+    @Ignore("Until the fuzzy logic of Noviflow bursting algorithm is clear")
+    def "Meter burst size should be #burstSize kbps if the flow rate is #flowRate kbps"() {
```

Comment:
Please specify that this is a non-centec case. I'm wondering: should we directly pick noviflow switches and have two types of tests: 'novi and centec' instead of 'centec and non-centec'. Thoughts?

Reply:
I'd rather keep splitting between Centec and non-Centec as OVS behaves similarly to the Novi.
We'll have to write separate test cases for Noviflow, however, as soon as Kilda starts making use of the OpenFlow 1.4 onwards.